PTGLgraphComputation/src/plcc/Main.java

```diff
@@ -5451,7 +5510,140 @@ private static long calculateSkipNeighborNum(Molecule mol1, Molecule mol2, int m
                     chainSkippedRes += (chainANumberResidues * chainBNumberResidues);
                 }
             }
+        }    
+        // - - - transitive contacts between chains (through ligand contacts) - - -
+        //
+
+        if (Settings.getBoolean("PTGLgraphComputation_transitive_contacts")){
+            // Iteration over all connections of each ligand atom
+            MolContactInfo currentMci = null;
+            Molecule resAOld = null;
+            Molecule resBOld = null;
+            Molecule resA;
+            Molecule resB;
+            Atom atomA;
+            Atom atomB;
+            Integer counterTransitiveContacts = 0;
+
+            for(ArrayList<Atom> currentTransitiveAtomList : ligandToProteinAtomContacts.values()){
+                // Check all keys to evaluate contacts
+
+                for(int a = 0; a < currentTransitiveAtomList.size(); a++){
+                    atomA = currentTransitiveAtomList.get(a);
+                    resA = atomA.molecule;
+                    for (int b = a+1; b < currentTransitiveAtomList.size(); b++){
+                        // Retrieve Residues of Atoms
+                        atomB = currentTransitiveAtomList.get(b);
+                        resB = atomB.molecule;
```

Comment:
In general, i, j and k are used as loop counters. If you want to refer to a and b because of resA and resB, I would suggest renaming a and b to aCounter and bCounter or something similar to put emphasis on the fact that a and b are not objects, for example atoms as in other functions, but counters.